src/init.cpp

```diff
@@ -688,14 +688,14 @@ void ThreadImport(std::vector<boost::filesystem::path> vImportFiles)
     {
         CImportingNow imp;
         int nFile = 0;
-        if (fReindexFast) uiInterface.InitMessage(_("Reindexing block headers from files..."));
         while (fReindexFast)
         {
             CDiskBlockPos pos(nFile, 0);
```

Comment:
Cannot move down after because `pos `is required by `GetBlockPosFilename`.

Reply:
What about an overload of function GetBlockPosFilename() which receives nFile instead of pos considering in that function the only relevant payload of pos is nfile ?

Reply:
Or even better ... Initialize only once outside the loop and for every loop reset nfile and pos.

Reply:
Yes, I have opted for constructing once outside loop and then resetting at the end of each iteration.